src/app/static/src/app/components/plots/Choropleth.vue

```diff
@@ -132,83 +123,40 @@
             currentFeatures() {
                 return this.featuresByLevel[this.detail]
             },
-            indicatorMetadata: function() {
+            indicatorMetadata: function () {
                 return this.choroplethIndicatorsMetadata.filter((i: IndicatorMetadata) => i.indicator == this.indicator)[0];
             },
-            options() {
-                const regionIndicators = this.regionIndicators;
-
-                if (!this.indicator){
-                    return {};
-                }
-
-                const indicator = this.indicator!!;
-                return {
-                    onEachFeature: function onEachFeature(feature: Feature, layer: Layer) {
-                        const area_id = feature.properties && feature.properties["area_id"];
-                        const area_name = feature.properties && feature.properties["area_name"];
-
-                        const values = regionIndicators[area_id];
-                        let value = values && values[indicator] && values[indicator]!!.value;
-                        const stringVal =  (value || value === 0) ? value.toString() : "";
-
-                        layer.bindPopup(`<div>
-                                <strong>${area_name}</strong>
-                                <br/>${stringVal}
-                            </div>`);
-                    }
-                }
-            },
             selectedRegionFeatures(): Feature[] {
-                if (this.selectedRegions && this.selectedRegions.length > 0) {
-                    return this.selectedRegions.map((r: NestedFilterOption) => this.getFeatureFromAreaId(r.id)!!);
-                } else if (this.countryFeature) {
-                    return [this.countryFeature];
+                if (this.selectedFilters.regions.length > 0) {
+                    return this.selectedFilters.regions.map((id: string) => this.getFeatureFromAreaId(id)!!);
                 }
-                return [];
+                return [this.countryFeature];
             }
         },
         data(): Data {
             return {
-                style: {
```

Comment:
Before I figured the proper bottlenecks out and was still just shaving off milliseconds here and there I moved the static bits of the style into the css. I don't think it made any difference to performance but it's a bit neater so I left it that way.